xbmc/favourites/FavouritesService.cpp

```diff
@@ -227,5 +284,14 @@ void CFavouritesService::GetAll(CFileItemList& items) const
 {
   CSingleLock lock(m_criticalSection);
   items.Clear();
-  items.Copy(m_favourites);
+  for (const auto& fav : m_favourites)
+  {
+    if (!IsMediasourceRestricted(fav))
+      items.Add(fav);
+  }
+}
+
+void CFavouritesService::RefreshFavourites()
+{
+  m_events.Publish(FavouritesUpdated{});
```

Comment:
You could check if anything has changed prior to sending the event. Then events would not be sent if not required. You have access to the current list of favourites, you would just need to reload them in order to check. Although now that I think about it this feels like more work than is really necessary!

Reply:
i believe the only not required call is here:
https://github.com/xbmc/xbmc/blob/1411dec80f5dbe2924dd09d7d11989f6f5819c1d/xbmc/interfaces/builtins/ProfileBuiltins.cpp#L72-L80

_user tries to enter master mode, but then they don't remember their code_
no big deal, i'll fix that